src/common.jl

```diff
@@ -72,6 +72,31 @@ nsamples(::Type{D}, x::AbstractMatrix) where {D<:Sampleable{Multivariate}} = siz
 nsamples(::Type{D}, x::Number) where {D<:Sampleable{Matrixvariate}} = 1
 nsamples(::Type{D}, x::Array{Matrix{T}}) where {D<:Sampleable{Matrixvariate},T<:Number} = length(x)
 
+for func in (:(==), :isequal, :isapprox)
+    @eval function Base.$func(s1::A, s2::B; kwargs...) where {A<:Sampleable, B<:Sampleable}
+        nameof(A) === nameof(B) || return false
+        fields = fieldnames(A)
+        fields === fieldnames(B) || return false
+
+        for f in fields
+            isdefined(s1, f) && isdefined(s2, f) || return false
+            $func(getfield(s1, f), getfield(s2, f); kwargs...) || return false
+        end
+
+        return true
+    end
+end
+
+function Base.hash(s::S, h::UInt) where S <: Sampleable
+    hashed = hash(nameof(S), h)
```

Comment:
I think it would be better to add a hash which is specific to the family of types that are being hashed, to avoid collisions with other types (even if it's unlikely). This is inspired by how `hash` works for `AbtractArray`:
```suggestion
    hashed = hash(Sampleable, h)
    hashed = hash(nameof(S), hashed)
```

Otherwise, looks good!